### PR TITLE
Enhanced RDoc for [] and []=

### PR DESCRIPTION
**Pull request by BurdetteLamar**
I had not covered the ground with previous RDoc for #[], so have revised that.  Both #[] and #[]= are complex, and need careful case-by-case treatment.

### CODE REVIEW

File: lib/csv/table.rb [hunk 1]
```diff
@@ -432,9 +462,41 @@ def headers
...
+    #
+    # If <tt>range.end</tt> is negative, calculates the ending index from the end:
+    #   table[0..-1] # => [["foo", "0"], ["bar", "1"], ["baz", "2"]]
+    #
+    # If <tt>range.start</tt> is negative, calculates the starting index from the end:
+    #   table[-2..2] # => [["foo", "0"], ["bar", "1"], ["baz", "2"]]
+    #
+    # If <tt>range.start</tt> is larger than <tt>table.size</tt>,
+    # returns an \Array of +nil+ values:
+    #   table[4..4] table[4..4]
```
Comment (kou) [inline lib/csv/table.rb:490]: Could you update the result value?

Comment (BurdetteLamar) [reply lib/csv/table.rb]: Thanks for catching this.  Fixed in new push.

**Comment by BurdetteLamar** [comment]
@kou, is this ready to merge?

**Comment by kou** [comment]
Thanks!

**Pull request merged**